### ISSUE TITLE
Error When Authenticating Through Facebook and Google

### ISSUE THREAD

**Issue by psopher** [firebase/FirebaseUI-iOS#86]
When I click the "Ok" button on the Facebook SDK or the "Allow" button on Google, the app crashes with this error:

    FValidation.m line 60
    +[FValidation validateFrom:validRootPathString:]
    Fatal Exception: InvalidPathValidation
    (child:) Must be a non-empty string and not contain '.' '#' '$' '[' or ']'

Sometimes the error says this instead:

    +[FValidation validateFrom:validPathString:]:

It occurs after pressing "OK" on the Facebook sign-in and after pressing "Allow" on the Google sign in. The error occurs before the `didSignInWithUser:` method on the `authViewController` can fire. Both Facebook and Google authentication are enabled in the Firebase dashboard.

This has to be a Firebase error. What's going on?

**Comment by chaogoog** [comment]
FValidation seems to be a class in the old FirebaseUI. Can you make sure that you're using the latest FirebaseUI libraries?

**Comment by psopher** [comment]
I am using most recent Firebase UI. I solved the issue by clearing the NSDocuments folder on launch, but I don't know why this solves the problem. Any ideas? It doesn't seem like the best way to do this

**Comment by psopher** [comment]
The issue was that I deleted everything in the `NSDocuments` folder in `applicationWillResignActive`. I was hoping just to delete images, but I ended up also deleting: "com.google.iid-keypair.plist", "g-checkin.plist, and "rmq2.sqlite"

This was an issue because when using Facebook or Google authentication, `applicationWillResignActive` is called. I fixed it by deleting `NSDocuments` files selectively and by doing it in the `applicationDidEnterBackground` method.
